### PR TITLE
[Resolve #1283] Deprecating iam_role, role_arn, and template_path; Introducing sceptre_role, cloudformation_service_role

### LINKED ISSUE #1283: iam_role and role_arn are confusing names that are easily confused.
### Subject of the issue
Right now, Sceptre has two very confusingly named StackConfigs that are easy to mistake for each other, despite operating very differently:

* `role_arn` is the ARN of the CloudFormation Service Role attached to a stack (permanently) and is assumed by _CloudFormation_ to perform all stack actions.
* `iam_role` is the ARN of the role _Sceptre_ assumes and uses to perform all stack actions.

**I believe we should deprecate these names and issue deprecation warnings when these configurations are used. Instead, we should transition to using much clearer names. I propose:
* `cloudformation_service_role`
* `sceptre_deployment_role`

Ideally, we'd deprecate the old configs (probably along with template_path), issue a warning, and then assign those values to the new fields. We could remove the deprecated fields on the next major version.

### PR DESCRIPTION
**Pull request by jfalkenstein**
This PR brings about three Stack Config deprecations (two of which have been renamed).

* `iam_role` has been deprecated and aliased to a new Stack Config key of `sceptre_role`. This change has also been made on the ConnectionManager.
* `role_arn` has been deprecated and aliased to a new Stack Config key of `cloudformation_service_role`.
* `template_path` has been deprecated officially (it has been discouraged in docs since 2.7 but this makes it official). Functionality on this hasn't really changed apart from the warnings (see below).

With these deprecations, **Sceptre will now emit DeprecationWarnings whenever these fields are set or accessed.** All locations within Sceptre have been fixed, but external and custom hooks and resolvers will also need to be updated if they're referencing these deprecated fields.

## PR Checklist

- [ ] Wrote a good commit message & description [see guide below].
- [ ] Commit message starts with `[Resolve #issue-number]`.
- [ ] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [ ] All unit tests (`make test`) are passing.
- [ ] Used the same coding conventions as the rest of the project.
- [ ] The new code passes pre-commit validations (`pre-commit run --all-files`).
- [ ] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [ ] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

File: sceptre/stack.py [hunk 1]
```diff
@@ -126,14 +143,25 @@ class Stack(object):
...
+        "iam_role", "sceptre_role", "4.0.0", "5.0.0"
+    )
+    role_arn = create_deprecated_alias_property(
+        "role_arn", "cloudformation_service_role", "4.0.0", "5.0.0"
+    )
+    sceptre_role_session_duration = None
+    iam_role_session_duration = create_deprecated_alias_property(
+        "iam_role_session_duration", "sceptre_role_session_duration", "4.0.0", "5.0.0"
+    )
+
```
Comment (jfalkenstein) [inline sceptre/stack.py:164]: These "deprecated alias properties" emit deprecation warnings and ultimately resolve to the correct values, both for getting and setting. You can find this function here: https://github.com/Sceptre/sceptre/pull/1295/files#diff-590afffaaf0b4f3a07656251e95e9b01e40c65fe8b5f8a9bf6cd84f4d3a1e5c4

File: docs/_source/docs/stack_config.rst [hunk 2]
```diff
@@ -24,8 +24,11 @@ particular Stack. The available keys are listed below.
 -  `parameters`_ *(optional)*
 -  `protected`_ *(optional)*
 -  `role_arn`_ *(optional)*
+-  `cloudformation_service_role`_ *(optional)*
 -  `iam_role`_ *(optional)*
```
Comment (zaro0508) [inline docs/_source/docs/stack_config.rst:28]: would it make sense to mark these as deprecated here as well?  something like `[Deprecated] iam_role_ *(optional)*`

**Review by zaro0508 — APPROVED** [review]
LGTM, much more clear.  thanks!

File: docs/_source/docs/stack_config.rst (hunk 2 above)
Comment (jfalkenstein) [reply docs/_source/docs/stack_config.rst:28]: I was thinking about that... that's easy enough to do.

File: docs/_source/docs/stack_config.rst [hunk 3]
```diff
@@ -312,49 +315,77 @@ role_arn
...
 
 .. warning::
 
-   If you set the value of ``iam_role_session_duration`` to a number that *GREATER* than 3600, you
-   will need to make sure that the ``iam_role`` has a configuration of ``MaxSessionDuration``, and
-   its value is *GREATER* than or equal to the value of ``iam_role_session_duration``.
-
-For more information on this configuration, its implications, and its uses, see
-:ref:`Sceptre and IAM: iam_role_session_duration <iam_role_permissions>`.
+   If you set the value of ``sceptre_role_session_duration`` to a number that *GREATER* than 3600, you
```
Comment (dboitnot) [inline docs/_source/docs/stack_config.rst:386]: Looks like a vestigial "that" here. I think this is supposed to read: "to a number *GREATER* than"

File: sceptre/stack.py (hunk 1 above)
Comment (dboitnot) [reply sceptre/stack.py:164]: That's pretty slick.

**Review by dboitnot — APPROVED** [review]
Didn't spot any issues.

**Pull request merged**